src/components/CustomerQuota/NoQuota/NoQuotaCard.test.ts

```diff
@@ -276,3 +277,125 @@ describe("NoQuotaCard utility functions", () => {
     });
   });
 });
+
+const allQuotaResponseNoAppeal: Quota | null = {
+  globalQuota: [
+    {
+      category: "appreciation-gift",
+      quantity: 0,
+      quotaRefreshTime: 1634800083079,
+      transactionTime: new Date("2020-10-21T07:08:03.079Z"),
+    },
+    {
+      category: "replacement-appreciation-gift",
+      quantity: 0,
+      quotaRefreshTime: 1634800102815,
+      transactionTime: new Date("2020-10-21T07:08:22.815Z"),
+    },
+  ],
+  localQuota: [
+    {
+      category: "appreciation-gift",
+      quantity: 9007199254740991,
+      quotaRefreshTime: 1634800083079,
+    },
+    {
+      category: "replacement-appreciation-gift",
+      quantity: 9007199254740991,
+      quotaRefreshTime: 1634800102815,
+    },
+  ],
+  remainingQuota: [
+    {
+      category: "appreciation-gift",
+      quantity: 0,
+      transactionTime: new Date("2020-10-21T07:08:03.079Z"),
+    },
+    {
+      category: "replacement-appreciation-gift",
+      quantity: 0,
+      transactionTime: new Date("2020-10-21T07:08:22.815Z"),
+    },
+  ],
+};
+
+const allProducts: CampaignPolicy[] | null = [
+  {
+    category: "appreciation-gift",
+    name: "🎁 Appreciation gift",
+    order: 1,
+    quantity: {
+      default: 1,
+      limit: 1,
+      period: 365,
+    },
+    type: "REDEEM",
+  },
+  {
+    category: "replacement-appreciation-gift",
+    categoryType: "APPEAL",
+    name: "🎁 Replacement",
+    order: 1,
+    quantity: {
+      default: 1,
+      limit: 1,
+      period: 365,
+    },
+    type: "REDEEM",
+  },
+];
+
+const allQuotaResponseHasAppeal: Quota | null = {
+  globalQuota: [
+    {
+      category: "appreciation-gift",
+      quantity: 0,
+      quotaRefreshTime: 1634803468038,
+      transactionTime: new Date("2020-10-21T08:04:28.038Z"),
+    },
+    {
+      category: "replacement-appreciation-gift",
+      quantity: 1,
+      quotaRefreshTime: 1634803476822,
+    },
+  ],
+  localQuota: [
+    {
+      category: "appreciation-gift",
+      quantity: 9007199254740991,
+      quotaRefreshTime: 1634803468038,
+    },
+    {
+      category: "replacement-appreciation-gift",
+      quantity: 9007199254740991,
+      quotaRefreshTime: 1634803476822,
+    },
+  ],
+  remainingQuota: [
+    {
+      category: "appreciation-gift",
+      quantity: 0,
+      transactionTime: new Date("2020-10-21T08:04:28.038Z"),
+    },
+    {
+      category: "replacement-appreciation-gift",
+      quantity: 1,
+    },
+  ],
+};
+
+describe("checkifAppealAvailable", () => {
```

Comment:
We usually name the test suite based on the function, so this should be "`checkHasAppealProduct`"

Reply:
looking at the function, i think we should test at least 3 things:
1) where allProducts has no appeal category type
2) where all product has appeal category type, and quota quantity for this category > 0
3) where all product has appeal category type, and quota quantity for this category == 0

I think you covered 3 and 2, but you should add another test case for 1!